nat-lab/tests/utils/connection_tracker.py

```diff
@@ -92,8 +102,10 @@ def __init__(
         self._connection: Connection = connection
         self._config: Optional[List[ConnectionTrackerConfig]] = configuration
         self._events: List[FiveTuple] = []
+        self._lock: Lock = threading.Lock()
 
         self._initialized: bool = False
+        self._synchronize: SynchronizeState = SynchronizeState.NOT_SYNCHRONIZED
```

Comment:
I would recommend to use https://docs.python.org/3/library/asyncio-sync.html#asyncio.Lock
or https://docs.python.org/3/library/asyncio-sync.html#asyncio.Event
This has a good explanation: https://stackoverflow.com/questions/25799576/whats-python-asyncio-lock-for

Reply:
There's just one downside of using `asyncio.Event`, is that it has no `timeout`, so it could get locked for ever.. or until the test fails after 120s.
[threading.Event](https://docs.python.org/3/library/threading.html#event-objects) on the other hand does have a `timeout`.

Reply:
you can always wrap async method with timeout(some awaitable) and get a timeout if you need

Reply:
timeout are never an issue in async code

Reply:
Wrapped the event.wait() method in timeout